commands/git/countcontributors.go

```diff
@@ -0,0 +1,340 @@
+			detailedRepos[repoName][authorEmail] = RepositoryDetailedSummary{
+				Email:      authorEmail,
+				LastCommit: lastCommit,
+			}
+		}
+	}
+}
+
+// aggregateUniqueContributors returns a list of unique contributors.
+// If a contributor has committed to multiple repositories, the most recent commit will be chosen.
+func (cc *CountContributorsCommand) aggregateUniqueContributors(uniqueContributors map[BasicContributor]Contributor) []Contributor {
+	// Choose the contributor with the most recent commit.
+	contributorsMap := make(map[string]Contributor)
+	for _, contributor := range uniqueContributors {
+		if _, exist := contributorsMap[contributor.Email]; !exist {
+			contributorsMap[contributor.Email] = contributor
+		} else if contributorsMap[contributor.Email].RepoLastCommit.Date < contributor.RepoLastCommit.Date {
+			contributorsMap[contributor.Email] = contributor
+		}
+	}
+	// Convert map into array.
+	var uniqueContributorsList []Contributor
+	for _, contributor := range contributorsMap {
+		uniqueContributorsList = append(uniqueContributorsList, contributor)
+	}
+	return uniqueContributorsList
+}
+
+func (cc *CountContributorsCommand) aggregateDetailedContributors(detailedContributors map[string]map[string]ContributorDetailedSummary) map[string][]ContributorDetailedSummary {
+	detailedContributorsList := make(map[string][]ContributorDetailedSummary)
+	for email, repos := range detailedContributors {
+		for _, detail := range repos {
+			detailedContributorsList[email] = append(detailedContributorsList[email], detail)
+		}
+	}
+	return detailedContributorsList
+}
+
+func (cc *CountContributorsCommand) aggregateDetailedRepos(detailedRepos map[string]map[string]RepositoryDetailedSummary) map[string][]RepositoryDetailedSummary {
+	detailedReposList := make(map[string][]RepositoryDetailedSummary)
+	for repo, authors := range detailedRepos {
+		for _, detail := range authors {
+			detailedReposList[repo] = append(detailedReposList[repo], detail)
+		}
+	}
+	return detailedReposList
+}
+
+func (cc *CountContributorsCommand) aggregateReportResults(uniqueContributors map[BasicContributor]Contributor, detailedContributors map[string]map[string]ContributorDetailedSummary, detailedRepos map[string]map[string]RepositoryDetailedSummary) Report {
+	report := Report{
+		TotalUniqueContributors: len(uniqueContributors),
+		ReportDate:              time.Now().Format(time.RFC3339),
+		NumberOfMonths:          fmt.Sprintf("%d", cc.MonthsNum),
+		UniqueContributorsList:  cc.aggregateUniqueContributors(uniqueContributors),
+	}
+
+	if cc.DetailedSummery {
+		report.DetailedContributorsList = cc.aggregateDetailedContributors(detailedContributors)
+		report.DetailedReposList = cc.aggregateDetailedRepos(detailedRepos)
+	}
+
+	return report
+}
+
+// Returns the Server details. The usage report is sent to this server.
+func (cc *CountContributorsCommand) ServerDetails() (*config.ServerDetails, error) {
```

Comment:
you can delete this func

Reply:
Have to implement it as part of the Command interface.